ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/PhoenixHBaseAccessor.java

```diff
@@ -1990,6 +1991,106 @@ public void saveMetricMetadata(Collection<TimelineMetricMetadata> metricMetadata
     return metadataMap;
   }
 
+  public List<TimelineMetricMetadata> scanMetricMetadataForWildCardRequest(Collection<String> metricNames,
+                                                                           String appId,
+                                                                           String instanceId) throws SQLException {
+    List<TimelineMetricMetadata> metadataList = new ArrayList<>();
+    Connection conn = getConnection();
+    PreparedStatement stmt = null;
+    ResultSet rs = null;
+
+    MetadataQueryCondition metadataQueryCondition = new MetadataQueryCondition(new ArrayList<>(metricNames), appId, instanceId);
```

Comment:
Minor nit: Why copy the collection into a new list here? Pass down reference of original collection instead.

Reply:
We have to do that since we need it to be a Set in the initial step, and a List further down the GET call flow.